### PR TITLE
Add build flag for GCC (for SDL: preventing GCC optimizing out security checks)

### PR DESCRIPTION
**Pull request by xipingyan**
### Details:
 - *Add build flag for GCC. Because some compiler flags restrict the compiler from making arbitrary decisions while handling undefined C/C++ behaviors.*

### Tickets:
 - 85228

### CODE REVIEW

**Review by peterchen-intel — APPROVED** [review]
As on the ticket 85228, there is not perf or accuracy gaps identified.

**Review by peterchen-intel — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
should not such flags be added to this file https://github.com/openvinotoolkit/openvino/blob/master/cmake/developer_package/compile_flags/sdl.cmake ? looks like it's a proper place, because such flags will be shared with private plugins and other modules which are built on top of OpenVINO Developer Package.

BTW, have you checked that a binary size is not increased when building with LTO enabled?

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,10 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "CMake build type" FORCE)
 endif()
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    add_compile_options(-fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv)
```
Comment (ilya-lavrenov) [inline CMakeLists.txt:21]: let's move it to https://github.com/openvinotoolkit/openvino/blob/master/cmake/developer_package/compile_flags/sdl.cmake with detailed comments why such options are needed

Comment (xipingyan) [reply CMakeLists.txt:21]: Ok, I will create a new PR.

Comment (xipingyan) [reply CMakeLists.txt:21]: hi @ilya-lavrenov , New PR: https://github.com/openvinotoolkit/openvino/pull/12143 has been created, please help review. Thanks.

**Pull request merged**